File: Library/src/main/java/org/endeavourhealth/library/model/imq/ECLStatus.java
package org.endeavourhealth.library.model.imq;

import lombok.Getter;
import lombok.Setter;

public class ECLStatus {

  @Getter
  @Setter
  private boolean valid;

  @Getter
  @Setter
  private Integer line;

  @Getter
  @Setter
  private Integer offset;

  @Getter
  @Setter
  private String message;
}

File: src/interfaces/ECLStatus.ts
import z from "zod";

// export interface ECLStatus {
//   valid?: boolean;
//   line?: number;
//   offset?: number;
//   message?: string;
// }

export const ECLStatusSchema = z.strictObject({
  valid: z.boolean().default(false),
  line: z.number().optional(),
  offset: z.number().optional(),
  message: z.string().optional()
});

export type ECLStatus = z.output<typeof ECLStatusSchema>;
